src/device-registry/utils/create-event.js

```diff
@@ -534,7 +535,7 @@ const createEvent = {
               });
               if (responseFromListEvents.success === true) {
                 let data = responseFromListEvents.data;
-                data[0].data = missingDataMessage ? [] : data[0].data;
+                data[0].data = !isEmpty(missingDataMessage) ? [] : data[0].data;
```

Comment:
In case there isn't any nearby Site to the provided GPS coordinates, just return an empty array of device measurements.